app/controllers/api/auth_key_pairs_controller.rb

```diff
@@ -1,5 +1,13 @@
 module Api
   class AuthKeyPairsController < BaseController
+    def create_resource(_type, _id = nil, data = {})
+      ext_management_system = ExtManagementSystem.find(data['ems_id'])
+      task_id = ManageIQ::Providers::CloudManager::AuthKeyPair.create_key_pair_queue(session[:userid], ext_management_system, data)
+      action_result(true, "Creating Cloud Key Pair #{data['name']} for Provider: #{ext_management_system.name}", :task_id => task_id)
```

Comment:
So the response is now a task_id. Assuming the original code returned the instance created in the DB, this could be a breaking change for existing consumers.

Reply:
Well, if an existing customer ever used this feature, it never actually worker for them. This is one of those issues where we mapped the HTTP payload into our DB but we never relayed the information towards the provider.

Reply:
My opinion, if it was broken we need to fix it, and if we break the API in doing so it was already not working

@Fryguy @abellotti what do you think?

Reply:
I agree with both @agrare @skateman comments. Perhaps we can work this out in the API meeting in a little while